MAINTAINERS.yml

```diff
@@ -0,0 +1,1166 @@
+    files:
+        - include/net/buf.h
+        - subsys/net/buf.c
+        - tests/net/buf/
+
+"Networking: CoAP":
+    status: maintained
+    maintainers:
+        - rveerama1
+    files:
+        - subsys/net/lib/coap/
+        - samples/net/sockets/coap_*/
+        - tests/net/lib/coap/
+
+"Networking: LWM2M":
+    status: maintained
+    maintainers:
+        - mike-scott
+    files:
+        - samples/net/lwm2m_client/
+        - subsys/net/lib/lwm2m/
+    labels:
+        - "area: LWM2M"
+
+"Networking: MQTT":
+    status: maintained
+    maintainers:
+        - rlubos
+    collaborators:
+        - jukkar
+    files:
+        - subsys/net/lib/mqtt/
+        - tests/net/lib/mqtt_packet/
+        - samples/net/mqtt_publisher/
+
+NIOS-2 arch:
+    status: maintained
+    maintainers:
+        - nashif
+    files:
+        - arch/nios2/
+        - include/arch/nios2/
+    labels:
+        - "area: NIOS2"
+
+nRF52 BSIM:
+    status: maintained
+    maintainers:
+        - aescolar
+    files:
+        - boards/posix/nrf52_bsim/
+    labels:
+        - "platform: nrf52_bsim"
+
+POSIX API layer:
+    status: maintained
+    maintainers:
+        - pfalcon
+    files:
+        - include/posix/
+        - lib/posix/
+        - tests/posix/
+    labels:
+        - "area: POSIX"
+
+Power management:
+    # Provisional entry, subject to approval
+    status: maintained
+    maintainer:
+        - wentongwu
+    files:
+        - include/power/power.h
+        - samples/subsys/power/
+        - subsys/power/
+    labels:
+        - "area: Power Management"
+
+RISCV arch:
+    # Provisional entry, subject to approval
+    status: orphaned
+    collaborators:
+        - mgielda
+        - nategraff-sifive
+    files:
+        - arch/riscv/
+        - include/arch/riscv/
+    labels:
+        - "area: RISCV"
+
+Sanitycheck:
+    status: maintained
+    maintainers:
+        - nashif
+    files:
+        - scripts/sanitycheck
+        - scripts/sanity_chk/
+    labels:
+        - "area: Sanitycheck"
+
+Settings:
+    status: maintained
+    maintainers:
+        - nvlsianpu
+    files:
+        - include/settings/
+        - subsys/settings/
+        - tests/subsys/settings/
+    labels:
+        - "area: Settings"
+
+Shell:
+    status: maintained
+    maintainers:
+        - jakub-uC
+    collaborators:
+        - carlescufi
+    files:
+        - include/shell/
+        - samples/subsys/shell/
+        - subsys/shell/
+        - tests/shell/
+        - tests/subsys/shell/
+    labels:
+        - "area: Shell"
+
+Shields:
+    status: maintained
+    maintainers:
+        - erwango
+    collaborators:
+        - avisconti
+        - jfischer-phytec-iot
+    files:
+        - boards/shields/
+    labels:
+        - "area: Shields"
+
+STM32:
+    status: maintained
+    maintainers:
+        - erwango
+    files:
+        - boards/arm/nucleo_*/
+        - boards/arm/stm32*_disco/
+        - boards/arm/stm32*_eval/
+        - drivers/*/*stm32*/
```

Comment:
Don‘t we create lots of conflicts with this entry?

Reply:
I'm happy to remove if this is not in line with general agreement and we expect there's only one assignee per PR.

Reply:
Yeap, I don't see a problem, I think @alexanderwachter might be referring to conflict for assignee-picking not conflicts in reviews.
when it comes to file/directory coverage, we do have conflicts allover the place, we just need to use some smart ways to pick assignees in PRs.

In this particular example CAN area and STM32 area both have maintainers appointed; for a PR changing an STM32 CAN drivers any of the respective maintainers may be assignees. We should be using some common sense here.
Otherwise, if a vendor area does not have an explicit maintainer, then Driver-Area maintainers become assignees for such PRs.

Reply:
It is just about the script. It would be annoying if the assignee picking is not working correctly. If we could have the maintainer as assignee and @erwango as reviewer, that would be perfect. But before such mechanism is working, we should be careful.

Reply:
Yeap, we should take care of this, thanks @alexanderwachter